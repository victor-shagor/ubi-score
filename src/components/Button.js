import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

const useStyles = makeStyles({
  btn: {
    backgroundColor: (props) => props.bgColor,
    height: "56px",
    width: (props) => props.width,
    color: "white",
    borderRadius: "5px",
    border: "none",
    fontFamily: "outfitMedium",
  },
});

const Button = ({ text, loading, onClick, ...props }) => {
  const Classes = useStyles(props);
  return (
    <button onClick={onClick} className={Classes.btn} type="submit">
      {loading ? <CircularProgress color="inherit" /> : text}
    </button>
  );
};

export default Button;
