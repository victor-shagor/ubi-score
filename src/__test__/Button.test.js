import Button from "components/Button";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

const mockProp = { text: "Name", width: "50px", bgColor: "blue" };

it("renders without crashing", () => {
  shallow(<Button />);
});

it("should render span", () => {
  const wrapper = mount(<Button {...mockProp} />);
  const value = wrapper.find("button").text();
  expect(value).toEqual("Name");
});

it("renders correctly", () => {
  const tree = shallow(<Button {...mockProp} />);
  expect(toJson(tree)).toMatchSnapshot();
});
