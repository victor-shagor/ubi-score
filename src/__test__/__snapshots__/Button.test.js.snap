// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="makeStyles-btn-1 makeStyles-btn-4"
  type="submit"
>
  Name
</button>
`;
